tests/AgDatabaseMove.Integration/Fixtures/TestAgDatabaseFixture.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace AgDatabaseMove.Integration.Fixtures
+{
+  using Config;
+  using System;
+  using System.Collections.Generic;
+  using Microsoft.SqlServer.Management.Smo;
+  using SmoFacade;
+  using System.Linq;
+
+  public class TestAgDatabaseFixture : IDisposable
+  {
+    public readonly TestAgDatabaseConfig _agConfig =
+      new TestConfiguration<TestAgDatabaseConfig>("TestAgDatabase")._config;
+
+    public readonly TestLoginConfig _loginConfig =
+      new TestConfiguration<TestLoginConfig>("TestLogin")._config;
+
+    public AgDatabase _agDatabase;
+
+    public IEnumerable<SmoFacade.Login> _createdLogins;
+
+    public TestAgDatabaseFixture()
+    {
+      _agDatabase = new AgDatabase(new DatabaseConfig
+      {
+        BackupPathSqlQuery = _agConfig.BackUpPathSqlQuery,
+        ConnectionString = _agConfig.ConnectionString,
+        DatabaseName = _agConfig.DatabaseName,
+        CredentialName = _agConfig.CredentialName
+      });
+      _agDatabase.AddLogin(new LoginProperties
+      {
+        Name = _loginConfig.LoginName,
+        Password = _loginConfig.Password,
+        LoginType = LoginType.SqlLogin,
+        DefaultDatabase = _loginConfig.DefaultDatabase
+      });
+
+      _createdLogins = GetCreatedLogins();
+    }
+
+    private IEnumerable<SmoFacade.Login> GetCreatedLogins()
+    {
+      List<SmoFacade.Login> logins = new List<SmoFacade.Login>();
+      logins.Add(_agDatabase._listener.Primary.Logins
+                   .SingleOrDefault(l => l.Name.Equals(_loginConfig.LoginName, StringComparison.InvariantCultureIgnoreCase)));
+      foreach (var server in _agDatabase._listener.Secondaries)
+      {
+        logins.Add(server.Logins
+                     .SingleOrDefault(l => l.Name.Equals(_loginConfig.LoginName, StringComparison.InvariantCultureIgnoreCase)));
+      }
+
+      return logins;
+    }
+
+    public void Dispose()
+    {
+      _agDatabase.DropLogin(new LoginProperties { Name = _loginConfig.LoginName });

```

Comment:
What happens if this line throws an exception when `Dispose()` is called from the destructor?

I guess in this context, it's not that big of an issue as it's just a test instance.
But I think it's something worth highlighting and something to be aware of.

https://stackoverflow.com/questions/1030455/how-to-handle-exception-thrown-from-dispose

In this instance, maybe we can clean up the logins explicitly instead after each test?

Reply:
yeah probably better to do this explicitly, or make it no-throw

Reply:
https://github.com/factset/AgDatabaseMove/blob/ad13f56971f5fa176ea523c0206de0a43e9a99e8/src/SmoFacade/Server.cs#L59-L62

I'm not too concerned about dropping throwing an exception here. The logic above should fail gracefully in the event the login doesn't exist. Maybe the database connection could give out before the everything gets disposed properly?